server/datastore/mysql/software.go

```diff
@@ -1721,3 +1721,306 @@ func (ds *Datastore) ListCVEs(ctx context.Context, maxAge time.Duration) ([]flee
 
 	return result, nil
 }
+
+func (ds *Datastore) ListHostSoftware(ctx context.Context, hostID uint, includeAvailableForInstall bool, opts fleet.ListOptions) ([]*fleet.HostSoftwareWithInstaller, *fleet.PaginationMetadata, error) {
+	// `status` computed column assumes that all results (pre, install and post)
+	// are stored at once, so that if there is an exit code for the install
+	// script and none for the post-install, it is because there is no
+	// post-install.
```

Comment:
Given what we've discussed at standup today, looks like that's fine, all results will be inserted at the same time. https://github.com/fleetdm/fleet/issues/18675